### PR TITLE
Replace custom signals to accept/reject

### LINKED ISSUE #3086: Use QDialog accept/reject instead of custom signals
## 🧰 Task
In looking through some settings stuff, I see this pattern being used in a few places, where a custom `valueChanged` signal is added to a `QDialog`, and emitted just before closing the dialog, based on whether the user clicked ok or cancel:

https://github.com/napari/napari/blob/d6bc683b019c4a3a3c6e936526e29bbd59cca2f4/napari/_qt/dialogs/qt_message_dialogs.py#L47-L59

usage example here:
https://github.com/napari/napari/blob/d6bc683b019c4a3a3c6e936526e29bbd59cca2f4/napari/_qt/dialogs/preferences_dialog.py#L191-L196

QDialogs run in `exec` mode (modal dialogs) already [have the concept of "accept" and "reject"](https://doc.qt.io/qt-5/qdialog.html#modal-dialogs), so this is reinventing the wheel a bit.  Instead of hooking up new internal signals to various potential outcomes, you can use the [`DialogCode`](https://doc.qt.io/qt-5/qdialog.html#DialogCode-enum) returned from the [`exec`](https://doc.qt.io/qt-5/qdialog.html#exec) method.

In this case, you can get rid of the whole `ConfirmDialog` class and do it with a single static dialog:

```py
ret = QMessageBox.question(
    None,
    "Dialog Title",
    "Are you sure you want to restore default settings?",
    QMessageBox.RestoreDefaults | QMessageBox.Cancel,
    QMessageBox.RestoreDefaults,
)
if ret == QMessageBox.RestoreDefault:
    _reset_widgets()
   # and then modify _reset_widgets to not take an event
```

cc @ppwadhwa

### PR DESCRIPTION
**Pull request by ppwadhwa**
I have replaced the custom ConfirmDialog with QMessageBox.  This will use functionality already provided by Qt.  

Closes #3086

### CODE REVIEW

**Comment by ppwadhwa** [comment]
While testing the new functionality, I realized a bug has been introduced at some point to the preferences dialog.  If I try to change the theme to light, and restore defaults, it will update the widget to "dark", but the console doesn't change themes.  I also noticed other bugginess to it, but I haven't narrowed down the cause. This was all on master, so I know it doesn't have to do with the changes here.

File: napari/_qt/dialogs/preferences_dialog.py [hunk 1]
```diff
@@ -190,14 +192,23 @@ def get_page_dict(self, field: 'ModelField'):
...
     def restore_defaults(self):
         """Launches dialog to confirm restore settings choice."""
-        self._reset_dialog = ConfirmDialog(
-            parent=self,
-            text=trans._("Are you sure you want to restore default settings?"),
+
+        restore_box = QMessageBox()
+
+        settings = get_settings()
+        restore_box.setStyleSheet(get_stylesheet(settings.appearance.theme))
```
Comment (tlambert03) [inline napari/_qt/dialogs/preferences_dialog.py]: is it still necessary to explicitly set the stylesheet if you give a parent to the QMessageBox?

```py
response = QMessageBox.question(
    self,
    trans._("Restore Settings"),
    trans._("Are you sure you want to restore default settings?"),
    QMessageBox.RestoreDefaults | QMessageBox.Cancel,
    QMessageBox.RestoreDefaults
)

if response == QMessageBox.RestoreDefaults:
    self._reset_widgets()
```

File: napari/_qt/widgets/qt_keyboard_settings.py [hunk 2]
```diff
@@ -119,16 +119,22 @@ def __init__(
...
 
-        self._reset_dialog = ConfirmDialog(
-            parent=self,
-            text=trans._(
-                "Are you sure you want to restore default shortcuts?"
-            ),
+        restore_box = QMessageBox()
+
+        settings = get_settings()
+        restore_box.setStyleSheet(get_stylesheet(settings.appearance.theme))
```
Comment (tlambert03) [inline napari/_qt/widgets/qt_keyboard_settings.py]: same as above...

**Comment by tlambert03** [comment]
> If I try to change the theme to light, and restore defaults, it will update the widget to "dark", but the console doesn't change themes. I also noticed other bugginess to it, but I haven't narrowed down the cause.

can you put this in code or a test?  I tried to reproduce it but wasn't able to figure it out

**Comment by ppwadhwa** [comment]
@tlambert03 I tried to reproduce programmatically, but haven't been able to yet.  Maybe its a disconnect with the widgets?  I need to spend some time looking at what is going on.  Here is a video of what I did that made me notice the problem.   If you do this, does it work for you on master?  
1. I reset all defaults.
2. Go to preferences and change theme to 'light'
3. click 'ok'  (though not sure this is necessary)
4. go back to preferences and restore defaults.  

In the final step, the defaults are reset on the widgets, but at least in the case of theme, the viewer is not changed.  The settings are also not saved for the next session.  When the widgets are reset, `get_settings.reset()` is being called, which helps the widget values to build properly to defaults. So, somewhere something is breaking. 

https://user-images.githubusercontent.com/54282105/128023672-fa1182b0-855c-4bf7-96e7-85a9a6ba5f50.mov

**Comment by tlambert03** [comment]
ah, thanks!

File: napari/_qt/dialogs/preferences_dialog.py (hunk 1 above)
Comment (ppwadhwa) [reply napari/_qt/dialogs/preferences_dialog.py]: @tlambert03 You're right, you don't need to do that.  setting the parent works great.  

Another question---If you use the QMessageBox.question function, you will get a symbol on the dialog--like this:
Screen Shot 2021-08-03 at 10 05 10 AM

Using a plain QMessageBox, you get no symbol:
Screen Shot 2021-08-03 at 10 06 44 AM

which do you like better?

Comment (tlambert03) [reply napari/_qt/dialogs/preferences_dialog.py]: I like the question (that's why used the static .question method), but don't have a strong preference, you can remove the icon if others prefer to

Comment (ppwadhwa) [reply napari/_qt/dialogs/preferences_dialog.py]: I will leave it with the question for now.  The code is more concise that way.  :)

**Comment by tlambert03** [comment]
fixed in #3123 ... (since settings manager is now nested, the issue was that "reset" was creating a new instance for the sub-model, thereby breaking connections). added a test so it shouldn't happen again

**Review by tlambert03 — APPROVED** [review]
🚀

**Review by sofroniewn — APPROVED** [review]
LGTM!

**Pull request merged**